src/Confluent.SchemaRegistry/CachedSchemaRegistryClient.cs

```diff
@@ -81,29 +81,22 @@ public CachedSchemaRegistryClient(IEnumerable<KeyValuePair<string, string>> conf
             // Convert.ToString returns "" in the null case here.
             var basicAuthSource = Convert.ToString(config.FirstOrDefault(prop => prop.Key.ToLower() == SchemaRegistryConfig.PropertyNames.SchemaRegistryBasicAuthCredentialsSource).Value);
             var basicAuthInfo = Convert.ToString(config.FirstOrDefault(prop => prop.Key.ToLower() == SchemaRegistryConfig.PropertyNames.SchemaRegistryBasicAuthUserInfo).Value);
-            if (basicAuthInfo != "" && basicAuthSource == "")
-            {
-                // default to USER_INFO if no source specified, since the strongly typed config
-                // class doesn't even expose the config source property.
-                basicAuthSource = "USER_INFO";
-            }
 
             string username = null;
             string password = null;
 
-            if (basicAuthSource == "USER_INFO")
+            if (basicAuthSource == "USER_INFO" || basicAuthSource == "")
             {
-                if (basicAuthInfo == "")
-                {
-                    throw new ArgumentException($"CachedSchemaRegistryClient: {SchemaRegistryConfig.PropertyNames.SchemaRegistryBasicAuthCredentialsSource} set to 'USER_INFO', but no value specified for {SchemaRegistryConfig.PropertyNames.SchemaRegistryBasicAuthUserInfo}.");
-                }
-                var userPass = (basicAuthInfo).Split(':');
-                if (userPass.Length != 2)
+                if (basicAuthInfo != "")
                 {
-                    throw new ArgumentException($"CachedSchemaRegistryClient: Configuration property {SchemaRegistryConfig.PropertyNames.SchemaRegistryBasicAuthUserInfo} must be of the form 'username:password'.");
+                    var userPass = (basicAuthInfo).Split(':');
```

Comment:
We should support escaping the :

Reply:
For the sake of documenting this - the RFC for basic auth disallows any colon in the username, and rather everything after the first colon should be considered to be a part of the password.

See #1767